app/models/user.rb

```diff
@@ -1,6 +1,18 @@
 class User < ActiveRecord::Base
   has_secure_password
 
+  before_validation :strip_whitespace
   validates :first_name, :last_name, :email, presence: true
   validates :email, uniqueness: true
+  validates :password, length: { minimum: 8 }
+  validates_format_of :email,
+    with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\Z/i
+
+  private
+
+  def strip_whitespace
+    self.first_name = self.first_name.strip if self.first_name
+    self.last_name = self.last_name.strip if self.last_name
```

Comment:
Looks good to me. Can't think of anything else to change here. The only thing that comes to mind is: Do we need "if self.first_name" and "if self.last_name" on lines 14,15 when we are already validating for presence on line 5?